src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -226,18 +228,23 @@ def from_toml(cls, lockfile: str | bytes) -> CoursierResolvedLockfile:
         """
 
         lockfile_str: str
+        lockfile_bytes: bytes
         if isinstance(lockfile, str):
             lockfile_str = lockfile
+            lockfile_bytes = lockfile.encode("utf-8")
         else:
             lockfile_str = lockfile.decode("utf-8")
+            lockfile_bytes = lockfile
 
         contents = toml.loads(lockfile_str)
         entries = tuple(
             CoursierLockfileEntry.from_json_dict(entry) for entry in (contents["entries"])
         )
+        metadata = JVMLockfileMetadata.from_lockfile(lockfile_bytes)
```

Comment:
This was written before I factored out the TOML-parsing code in #14175, this will fail if the metadata header block is not in place.

Reply:
Unsure if we want to add a super-quick deprecation cycle here, but I don't believe we've put the contents of #14175 in a release yet?

Reply:
We haven't, no need to handle deprecation.